### PR TITLE
Use a single model for stroke and fill in ColumnLayer

### PR DESCRIPTION
**Pull request by Pessimistress**
For #3015 

#### Change List
- Remove unused options, reduce attribute size (1/3 vertex count)
- Move custom vertices handling into `ColumnGeometry` constructor

### CODE REVIEW

**Comment by Pessimistress** [comment]
@ibgreen To put this change in context - I agree that the few bytes saved by using TRIANGLE_STRIP may not worth the complexity, but I do want to avoid having to use multiple models for fill and stroke in this layer (see https://github.com/uber/deck.gl/pull/3098). Aside from the duplication of resources, many of the base layer's default implementations cannot be used in multi-model layers, leading to hard-to-manage edge cases.

What I did in here is still a hack (manually setting the `isIndexed` prop of the model to use/ignore the indices). Ideally the Model class could allow me to tie draw modes to different index arrays and switch between them.

**Comment by ibgreen** [comment]
> What I did in here is still a hack (manually setting the isIndexed prop of the model to use/ignore the indices). Ideally the Model class could allow me to tie draw modes to different index arrays and switch between them.

OK thanks. Sounds like our primitive geometries should provide the option to generate wireframe geometries (which as you say can really just be a second set of indices to the same vertices)?

> What I did in here is still a hack (manually setting the isIndexed prop of the model to use/ignore the indices). Ideally the Model class could allow me to tie draw modes to different index arrays and switch between them.

I do agree that if you are going systematically down this road it would make sense to do something like that. But that seems like an overly specialized feature for in luma.gl. I think that could be something that should be prototyped in @tsherif 's "engine" first, before (possibly) moving it to `Model`.

I would imagine a model/geometry manager class that lets you define abstract modes `wireframe`, `solid` and map them to draw mode and indices, rather than use draw mode as key. In this example for instance, you could easily revert to `GL.TRIANGLES` without changing the logic, so better avoid keying on draw mode.

File: modules/layers/src/column-layer/column-geometry.js [hunk 1]
```diff
@@ -1,136 +1,114 @@
...
-        indices[index + 3] = vertsAroundEdge * (i + 0) + 0 + j;
-        indices[index + 4] = vertsAroundEdge * (i + 1) + 1 + j;
-        indices[index + 5] = vertsAroundEdge * (i + 1) + 0 + j;
-      }
-    }
+    indices = null;
   }
 
   return {
     indices,
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js]: See follow up PR - this geometry is reused for both draw modes.

File: modules/layers/src/column-layer/column-geometry.js [hunk 2]
```diff
@@ -1,136 +1,114 @@
...
     attributes: {
       POSITION: {size: 3, value: positions},
-      NORMAL: {size: 3, value: normals},
-      TEXCOORD_0: {size: 2, value: texCoords}
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/layers/src/column-layer/column-geometry.js]: In the `H3HexagonLayer` case, this geometry is regenerated when viewport changes, and creating new Buffer objects for each attribute. I would like to keep it minimal until we have a better solution.

File: modules/layers/src/column-layer/column-geometry.js [hunk 3]
```diff
@@ -1,136 +1,110 @@
...
+
+  // side tesselation: 0, 1, 2, 3, 4, 5, ...
+  //
+  // 0 - 2 - 4  ... top
+  // | / | / |
+  // 1 - 3 - 5  ... bottom
+  //
+  for (let j = 0; j < vertsAroundEdge; j++) {
+    const a = j * stepAngle;
+    const vertex = vertices && vertices[j % nradial];
```
Comment (xintongxia) [inline modules/layers/src/column-layer/column-geometry.js]: Add comments about when vertex could be null?

**Pull request merged**